app/Http/Controllers/VisitTransfer/Admin/Application.php

```diff
@@ -111,6 +112,18 @@ public function postAccept(ApplicationAcceptRequest $request, ApplicationModel $
                        ->withSuccess('Application #'.$application->public_id.' - '.$application->account->name.' accepted &amp; candidate notified.');
     }
 
+    public function postComplete(ApplicationCompleteRequest $request, ApplicationModel $application)
+    {
+        try {
+            $application->complete(Input::get('complete_staff_note', null), Auth::user());
+        } catch (\Exception $e) {
+            return Redirect::back()->withError($e->getMessage());
```

Comment:
This blanket try catch for all Exceptions would stop us from being able to log any errors as a result of the complete method. Do we even need the try catch at all? (Also if we have a database error or likewise it may potentially give the user information they are not meant to see)

Reply:
I've implemented this how its been done for Reject and Accept, if we want to change this I would suggest changing it for the others too?

Reply:
I would say so. Thoughts @NFarrington ?

Reply:
The inconsistency in how we implement the methods that throw these exceptions means that more work is required than the scope of this PR - as the redirect provides the exception message to the user it's not a massive deal.

However, it should be easy to change the blanket Exception to the actual exceptions that need catching, rather than all exceptions, so that should be done for all 3 in this PR.